### PR TITLE
feat: Update substrate to v1.1.0

### PR DESCRIPTION
**Pull request by ukint-vs**
- move to a new repo https://github.com/gear-tech/polkadot-sdk 
- some imports moved from `frame-support` to `sp-runtime`
- update election-provider config
- update `RuntimeApi`
- `try-runtime-cli` removed from `gear-cli` use https://github.com/paritytech/try-runtime-cli instead

todo:

- [x]  Update mocks

@reviewer-or-team

### CODE REVIEW

File: common/src/gas_provider/mod.rs [hunk 1]
```diff
@@ -17,9 +17,8 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 use super::*;
-use frame_support::{
-    sp_runtime::traits::Zero, traits::tokens::Balance as BalanceTrait, RuntimeDebug,
-};
+use frame_support::{sp_runtime::traits::Zero, traits::tokens::Balance as BalanceTrait};
+use sp_runtime::RuntimeDebug;
```
Comment (shamilsan) [inline common/src/gas_provider/mod.rs:21]: Isn't is better to import trait `Zero` directly from `sp_runtime`?

```suggestion
use frame_support::{traits::tokens::Balance as BalanceTrait};
use sp_runtime::{traits::Zero, RuntimeDebug};
```

File: Cargo.toml [hunk 2]
```diff
@@ -249,125 +249,125 @@ sandbox-wasmer = { package = "wasmer", version = "2.2", features = ["singlepass"
...
-substrate-build-script-utils = { version = "3.0.0", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
-substrate-frame-rpc-system = { version = "4.0.0-dev", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
-substrate-rpc-client = { version = "0.10.0-dev", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
-substrate-state-trie-migration-rpc = { version = "4.0.0-dev", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
-substrate-test-client = { version = "2.0.0", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
-substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
-try-runtime-cli = { version = "0.10.0-dev", git = "https://github.com/gear-tech/substrate.git", branch = "gear-polkadot-v1.0.0-canary" }
+frame-benchmarking = { version = "4.0.0-dev", git = "https://github.com/gear-tech/polkadot-sdk.git", branch = "gear-release-polkadot-v1.1.0", default-features = false }
```
Comment (breathx) [inline Cargo.toml:252]: Maybe simplify deps name?
```suggestion
frame-benchmarking = { version = "4.0.0-dev", git = "https://github.com/gear-tech/polkadot-sdk.git", branch = "gear-v1.1.0", default-features = false }
```

File: common/src/storage/primitives/map.rs [hunk 3]
```diff
@@ -21,8 +21,7 @@
 //! This primitive defines interface of interaction
 //! with globally stored single-key map (Key -> Value).
 
-use frame_support::codec::{Encode, EncodeAppend, EncodeLike};
-
+use sp_runtime::codec::{Encode, EncodeAppend, EncodeLike};
```
Comment (breathx) [inline common/src/storage/primitives/map.rs:24]: here and in similar places directly from codec mb?
btw was there any reason to switch the dep source? frame_support no longer re-imports these?

File: node/testing/src/client.rs [hunk 4]
```diff
@@ -56,7 +56,7 @@ pub type Client = client::Client<
 >;
 
 /// Transaction for given runtime.
-pub type Transaction = sc_client_api::backend::TransactionFor<Backend, runtime_primitives::Block>;
+// pub type Transaction = sc_client_api::backend::TransactionFor<Backend, runtime_primitives::Block>;
```
Comment (breathx) [inline node/testing/src/client.rs:59]: could be removed at all

File: pallets/gear/src/benchmarking/mod.rs [hunk 5]
```diff
@@ -75,11 +75,11 @@ use core_processor::{
     configs::{BlockConfig, PageCosts, TESTS_MAX_PAGES_NUMBER},
     Ext, ProcessExecutionContext, ProcessorContext, ProcessorExternalities,
 };
+
+use parity_scale_codec::Encode;
+
```
Comment (breathx) [inline pallets/gear/src/benchmarking/mod.rs:80]:
```suggestion
use parity_scale_codec::Encode;
```

File: pallets/staking-rewards/src/mock.rs [hunk 6]
```diff
@@ -233,6 +233,9 @@ parameter_types! {
     pub const MaxElectableTargets: u16 = 10_000;
     pub const MaxOnChainElectingVoters: u32 = 500;
     pub const MaxOnChainElectableTargets: u16 = 100;
+    // pub static ElectionsBounds: ElectionBounds = ElectionBoundsBuilder::default().build();
```
Comment (breathx) [inline pallets/staking-rewards/src/mock.rs:236]: deadcode

**Review by breathx — APPROVED** [review]

File: common/src/storage/primitives/map.rs (hunk 3 above)
Comment (ukint-vs) [reply common/src/storage/primitives/map.rs:24]: > frame_support no longer re-imports these?
Yes

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -131,10 +131,19 @@ jobs:
...
-          cargo build -p gear-cli --features try-runtime --locked ${{ matrix.profile_flags }}
-          ./target/${{ matrix.profile }}/gear try-runtime --runtime ./target/${{ matrix.profile }}/wbuild/vara-runtime/vara_runtime.wasm on-runtime-upgrade --checks live --uri ws://rpc-private.vara-network.io:9944
+          export RUST_LOG=remote-ext=debug,runtime=debug
+
+          echo "---------- Downloading try-runtime CLI ----------"
+          curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/v0.5.4/try-runtime-x86_64-unknown-linux-musl -o try-runtime
+          chmod +x ./try-runtime
+
+          echo "---------- Building vara runtime ----------"
+          time cargo build -p gear-cli --features try-runtime --locked ${{ matrix.profile_flags }}
```
Comment (ukint-vs) [inline .github/workflows/build.yml:141]:
```suggestion
          time cargo build -p vara-runtime --features try-runtime --locked ${{ matrix.profile_flags }}
```

Comment (ukint-vs) [reply .github/workflows/build.yml:141]: Build runtime

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -131,10 +131,19 @@ jobs:
...
-          cargo build -p gear-cli --features try-runtime --locked ${{ matrix.profile_flags }}
-          ./target/${{ matrix.profile }}/gear try-runtime --runtime ./target/${{ matrix.profile }}/wbuild/vara-runtime/vara_runtime.wasm on-runtime-upgrade --checks live --uri ws://rpc-private.vara-network.io:9944
+          export RUST_LOG=remote-ext=debug,runtime=debug
+
+          echo "---------- Downloading try-runtime CLI ----------"
+          curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/v0.5.4/try-runtime-x86_64-unknown-linux-musl -o try-runtime
+          chmod +x ./try-runtime
+
+          echo "---------- Building vara runtime ----------"
+          time cargo build -p vara-runtime --features try-runtime --locked ${{ matrix.profile_flags }}
```
Comment (breathx) [inline .github/workflows/build.yml:141]:
```suggestion
          time cargo build -p vara-runtime -F try-runtime,dev --locked ${{ matrix.profile_flags }}
```

**Comment by breathx** [comment]
migrations behaviour should be fixed

File: pallets/gear-messenger/src/migrations.rs [hunk 9]
```diff
@@ -219,8 +233,7 @@ mod v2 {
     }
 
     #[cfg(feature = "try-runtime")]
-    #[allow(type_alias_bounds)]
-    pub type DispatchStash<T: Config> = StorageMap<
+    pub type DispatchStashV2<T> = StorageMap<
```
Comment (breathx) [inline pallets/gear-messenger/src/migrations.rs:236]: why?

File: pallets/gear-messenger/src/migrations.rs [hunk 10]
```diff
@@ -85,30 +86,43 @@ impl<T: Config> OnRuntimeUpgrade for MigrateToV3<T> {
...
+            let mut count = v2::Waitlist::<T>::iter().count();
+            count += v2::Dispatches::<T>::iter().count();
+            count += v2::DispatchStashV2::<T>::iter().inspect(
+                |store| {
+                    if store.1.0.context.is_some() {
+                        panic!("Previous context on StoredDispatch in DispatchStash should always be None, but was Some for message id {:?}", store.1.0.message.id());
+                    }
+                },
+            ).count();
+            Ok((count as u64).encode())
```
Comment (breathx) [inline pallets/gear-messenger/src/migrations.rs:100]: dont you want to return here encoded Option<u64>?

**Review by breathx — APPROVED** [review]

File: pallets/gear-messenger/src/migrations.rs (hunk 10 above)
Comment (ukint-vs) [reply pallets/gear-messenger/src/migrations.rs:100]: Simple way

**Pull request merged**